### PR TITLE
Update expo image loader to remove Fresco and update Glide

### PR DESCRIPTION
**Pull request by josephyanks**
# Why

Attempting to update dependencies that necessitate the use of jetifier.

The expo-image-loader uses both Fresco and Glide, which both accomplish the same thing.

# How

* Updated `com.github.bumptech.glide:glide` from `4.9.0` to `4.15.1` and added a variable in `ext` to customize the glide version in the future.
* Removed `com.facebook.fresco:fresco` from the project
* Replaced `loadImageForDisplayFromURL` method body with Glide implementation
* Replaced deprecated `SimpleTarget` with `CustomTarget`
* Removed `NonNull` and `Nullable` annotations from Kotlin code, as they are lint violations (nullity is enforced by language - specifically by contract with `ImageLoaderInterface`)

# Test Plan

There are no unit tests associated with this project, so I tested the changed method `loadImageForDisplayFromURL` on my own project.

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [X] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by josephyanks** [comment]
@Kudo not sure why no reviewers got added but wanted to make sure someone saw this - I believe this is the last of the jetified dependencies from expo 🤞🏻

**Comment by Kudo** [comment]
@lukmccall could you help to review this pr when you get a chance?

**Review by lukmccall — CHANGES_REQUESTED** [review]
Everything seems great! However, we need to use a different version of Glide. Kindly use Glide version `4.13.2`, as it is already being utilized in the expo-image and cannot be updated at the moment.

**Comment by lukmccall** [comment]
> Updated. Out of curiosity, why can't glide be updated past 4.13.2? Wondering if there's something I should worry about in our project

We're using an external decoder for some file formats, which doesn't seem to work with the newest version.

File: packages/expo-image-loader/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 ## Unpublished
 
+- Removed `com.facebook.fresco:fresco` dependency ([#22542](https://github.com/expo/expo/pull/22542) by [@josephyanks](https://github.com/josephyanks))
+-
```
Comment (lukmccall) [inline packages/expo-image-loader/CHANGELOG.md:6]:
```suggestion

```

**Review by lukmccall — APPROVED** [review]

**Comment by fobos531** [comment]
@lukmccall Recently a new version of the WebP decoder (which I assume was blocking the version upgrade) got released: https://github.com/zjupure/GlideWebpDecoder/commit/a57b21c763f8ad2325d29001ed6fab490cdb530e

That might be worth looking at.

**Pull request merged**